Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2926,8 +2926,23 @@ extension BrowserViewController: IntroViewControllerDelegate {
         } else {
             let signInVC = FxAContentViewController(profile: profile)
             signInVC.delegate = self
-            signInVC.url = profile.accountConfiguration.signInURL
-            signInVC.fxaOptions = fxaOptions
+            
+            // Append any passed query strings to the signin url. Note that you can't
+            // override the service and context params.
+            if fxaOptions.query != nil {
+                var url = "";
+                for param in fxaOptions.query! {
+                    if (param.key == "service" || param.key == "context") {
+                        continue;
+                    }
+                    url = url + "&" + param.key + "=" + param.value
+                }
+                let newUrl = URL(string: "\(profile.accountConfiguration.signInURL)\(url)")
+                signInVC.url = newUrl
```

Comment:
The url on the VC should really be private (thats a different matter) But better to do this in the initialization or via another public method so we can test this.